### ISSUE TITLE
md5sum not found on OS X

### ISSUE THREAD

**Issue by zmerlynn** [GoogleCloudPlatform/kubernetes#6389]
Filing for @roberthbailey 

This code is wrong https://github.com/GoogleCloudPlatform/kubernetes/blob/master/build/common.sh#L595: 
```
      local md5_sum=$(md5sum ${binary_file_path} | awk '{print $1}')
```

The local needs to be on a separate line from the command execution, and we need to be resilient to `md5sum` not existing on OS X, like: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/build/common.sh#L263

**Comment by roberthbailey** [comment]
For once I was actually watching the output of a build and noticed that I was seeing an error about md5sum not being found. I'm building on a mac using boot2docker. 

The short_hash function (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/build/common.sh#L256) works fine on my mac as md5 does exist.

**Comment by zmerlynn** [comment]
Actually, I'm just going to fix this as a follow-on to #6369.

**Comment by ArtfulCoder** [comment]
@zmerlynn Thank you!
